MediaPlayer/MediaPlayer.js

```diff
@@ -391,16 +489,133 @@ const MediaPlayerBehaviorDecorator = hoc((config, Wrapped) => { // eslint-disabl
 		 * @memberof agate/MediaPlayer.MediaPlayerBase.prototype
 		 * @public
 		 */
-		loopChange = () => {
+		handleOnRepeat = () => {
+			// Handling the 3 states of repeat: repeat none, repeat one and repeat all
+			switch (this.state.repeat) {
+				case 'none':
+					this.setState(prevState  => {
+						return ({loop: !prevState.loop});
+					}, () => {
+						this.media.loop = this.state.loop;
+					});
+					this.setState({repeat: 'one'});
+					break;
+				case 'one':
+					this.setState({repeat: 'all'});
+					this.media.loop = false;
+					break;
+				case 'all':
+					this.setState({repeat: 'none'});
+					this.media.loop = false;
+			}
```

Comment:
How about simplifying this logic to below?

```suggestion
			let loop = false;

			this.setState(({repeat}) => {
				switch (repeat) {
					case 'none':
						loop = !repeat.loop;
						return ({loop, repeat: 'one'});
					case 'one':
						return ({repeat: 'all'});
					case 'all':
						return ({repeat: 'none'});
				}
			}, () => {
				this.media.loop = loop;
			});
```